docs/consensus_decisions.md

```diff
@@ -0,0 +1,69 @@
+We need to decide on when the validation can and should happen, including the meaning of “build time” and “run time” in regards to validation.
+
+## 3: Allow for selectors to select a case depending on the value of one or more input arguments.
+
+**Discussion:**
+This is a prerequisite for top-level selectors to be able to represent complex messages, without requiring those messages to be split up in an unergonomic manner.
+This is an extension or relaxation of what's allowed in MessageFormat 1.
+
+While message references make it technically possible for the data model to represent multi-argument selectors otherwise, this requires the use of n²-1 artificial "messages", where n is the number of arguments. This is not desirable.
+
+## 4: Only allow for selectors at the top level of a message.
+
+**Discussion:**
+Requiring selectors to only be available at the top level is a good way of helping to maintain the translatability of messages, as well as otherwise guiding MessageFormat 2 users towards good practices.
+
+After an in-depth exploration of the problem space, we have determined that while selectors are a necessary feature of MessageFormat, it is not necessary for them to be available within the body of a message, or directly within a case of a parent selector.
+
+All identified use cases of such constructions may be cleanly represented using a top-level selector that may use more than one input argument to select among a set of messages.
+Furthermore, we may enable complete reversibility of message transformations to and from languages such as MessageFormat 1 and Fluent by using message references.
+
+## 5: Top level selectors together with message references provides the same value as nested selectors at a lower cost.
+
+**Discussion:**
+Nested selectors provides capabilities that may be useful in avoiding variant permutation explosion in edge cases, but the use of them has not been evaluated in production localization systems to date.
+The group believes that the known value of this feature can be sufficiently covered by the combination of message references and top level selection features, which together provide a sufficient feature set at a lower cost to the ecosystem than nested selectors would do.
+
+## 6: The model will be designed to leave the door open for nested selectors being potentially reconsidered in the future.
+
+**Discussion:**
+The cost analysis of the nested selectors feature was performed in the absence of sufficient in-field experience of use in production systems.
+In result, the group's decision to not currently incorporate the feature is based on the lack of sufficient known value that would require them, which the group recognizes may change in the future.
+In result, it is the intent of the group to attempt to design MessageFormat 2 in such a way, that wouldn't block future revisions of the standard to be extended with nested selectors feature.
```

Comment:
```suggestion
In result, it is the intent of the group to design MessageFormat 2 in a way that wouldn't prevent future revisions of the standard to be extended with nested selectors feature.
```

Reply:
Yeah, this is a better fit for the updated language.